src/sql/workbench/parts/connection/common/connectionProviderExtension.ts

```diff
@@ -13,12 +14,35 @@ import { deepClone } from 'vs/base/common/objects';
 import * as azdata from 'azdata';
 import * as resources from 'vs/base/common/resources';
 
+
 export interface ConnectionProviderProperties {
 	providerId: string;
 	displayName: string;
 	connectionOptions: azdata.ConnectionOption[];
 }
 
+class ResolveChecker {
+	private resolvedUsers = new Map<IExtensionDescription, boolean>();
```

Comment:
Why is this a whole separate class? It seems easiest to just have a set of resolved names and check against that

Reply:
Just was following the structure of the existing classes, I agree its a bit unnecessary.